rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -161,10 +182,20 @@ def __init__(self, socket, msg_router):
             content=message_content,
             message_type=message_type)
 
-        try:
-            await self._socket.send_multipart([message.SerializeToString()])
-        except asyncio.CancelledError:
-            raise
+        # Send the message. Backoff and retry in case of an error
+        backoff = _Backoff(max_retries=3,
+                           interval=200,
+                           error=SendBackoffTimeoutError())
+
+        while True:
+            try:
+                await self._socket.send_multipart(
+                    [message.SerializeToString()])
+                break
+            except asyncio.CancelledError:
+                raise
+            except zmq.error.Again:
```

Comment:
If this contains an interesting string (which seems possible... this is generated by EAGAIN, so it might be very generic), then it should be logged with the other info (see above).